### PR TITLE
[v2.9] Implemented a go lib that will post the qase/go failures to our custom webhook on slack.

### PR DESCRIPTION
**Pull request by igomez06**
## Problem
 Needed a way to report recurring run failures to slack
 
## Solution
 Created a custom webhook and new slack channel. 
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Comment by igomez06** [comment]
Backports:
- https://github.com/rancher/rancher/pull/45822
- https://github.com/rancher/rancher/pull/45823

**Review by floatingman — APPROVED** [review]
LGTM

File: tests/v2/validation/pipeline/slack/slack.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		Text: Text{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("Failures: %s", testRunName),
+		},
+	}
+
+	blockSlice = append(blockSlice, titleBlock)
+	for _, testCase := range testCaseSlice {
+		combinedTestSuite := strings.Join(testCase.TestSuite, "/")
+		// var testSuiteBlock Block
```
Comment (caliskanugur) [inline tests/v2/validation/pipeline/slack/slack.go:47]: I think this is from dev stage

**Review by caliskanugur — COMMENTED** [review]
Looks good to me, I've got a single nit

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Pull request merged**